pkg/render/egressgateway/egressgateway.go

```diff
@@ -0,0 +1,404 @@
+			Template: *c.deploymentPodTemplate(),
+		},
+	}
+	return &d
+}
+
+func (c *component) deploymentPodTemplate() *corev1.PodTemplateSpec {
+	var ps []corev1.LocalObjectReference
+	for _, x := range c.config.PullSecrets {
+		ps = append(ps, corev1.LocalObjectReference{Name: x.Name})
+	}
+	ptSpec := &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: c.egwBuildAnnotations(),
+			Labels:      c.config.EgressGW.Spec.Template.Metadata.Labels,
+		},
+		Spec: corev1.PodSpec{
+			ImagePullSecrets:              ps,
+			Affinity:                      c.config.EgressGW.GetAffinity(),
+			TopologySpreadConstraints:     c.config.EgressGW.GetTopoConstraints(),
+			NodeSelector:                  c.config.EgressGW.GetNodeSelector(),
+			TerminationGracePeriodSeconds: c.config.EgressGW.GetTerminationGracePeriod(),
+			InitContainers:                []corev1.Container{*c.egwInitContainer()},
+			Containers:                    []corev1.Container{*c.egwContainer()},
+			Volumes:                       []corev1.Volume{*c.egwVolume()},
+		},
+	}
+	if c.config.Provider == operatorv1.ProviderRKE2 {
+		ptSpec.Spec.ServiceAccountName = "tigera-egress-gateway"
+	}
+	return ptSpec
+}
+
+func (c *component) egwBuildAnnotations() map[string]string {
+	annotations := map[string]string{}
+	annotations["cni.projectcalico.org/ipv4pools"] = c.getIPPools()
+	if c.config.EgressGW.Spec.AWS != nil && len(c.config.EgressGW.Spec.AWS.ElasticIPs) > 0 {
+		annotations["cni.projectcalico.org/awsElasticIPs"] = c.getElasticIPs()
+	}
+	return annotations
+}
+
+func (c *component) egwInitContainer() *corev1.Container {
+	initContainerPrivileges := true
+	return &corev1.Container{
+		Name:            "egress-gateway-init",
+		Image:           c.config.egwImage,
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Command:         []string{"/init-gateway.sh"},
+		SecurityContext: &corev1.SecurityContext{Privileged: &initContainerPrivileges},
+		Env:             c.egwInitEnvVars(),
+	}
+}
+
+func (c *component) egwContainer() *corev1.Container {
+	return &corev1.Container{
+		Name:            "egress-gateway",
+		Image:           c.config.egwImage,
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Env:             c.egwEnvVars(),
+		Resources:       c.getResources(),
+		VolumeMounts:    c.egwVolumeMounts(),
+		Ports:           c.egwPorts(),
+		Command:         []string{"/start-gateway.sh"},
+		ReadinessProbe:  c.egwReadinessProbe(),
+		SecurityContext: c.egwSecurityContext(),
+	}
+}
+
+func (c *component) egwReadinessProbe() *corev1.Probe {
+	return &corev1.Probe{
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: "/readiness",
+				Port: intstr.FromInt(int(DefaultHealthPort)),
+			},
+		},
+		InitialDelaySeconds: 3,
+		TimeoutSeconds:      1,
+		SuccessThreshold:    1,
+		PeriodSeconds:       3,
+	}
+}
+
+func (c *component) egwSecurityContext() *corev1.SecurityContext {
+	return &corev1.SecurityContext{
+		Capabilities: &corev1.Capabilities{
```

Comment:
Would it be possible to specify the other fields as well?

Especially: RunAsUser, RunAsGroup, RunAsNonRoot, Privileged

For other EE components we are normalizing all of the users and groups to 10001:10001. It would be nice  if EGW would do the same.